src/compressed_tensors/utils/match.py

```diff
@@ -46,23 +49,70 @@ def match_named_modules(
     :param targets: target strings, potentially containing "re:" prefixes
     :param ignore: targets to ignore, potentially containing "re:" prefixes
     :param warn_on_fail: if True, warns if any targets do not match any modules in model
+    :param warn_on_unmatched_ignores: if True, warns if any ignores do not match any modules in model
+    :param yield_matched_targets: if True, yields the matched targets in addition to the module name and module
```

Comment:
I'd rather not over-complicate this utility. Is it possible to achieve what we want without modifying `match_named_modules`?

We might need to break the schemes out, similar to [here](https://github.com/neuralmagic/compressed-tensors/blob/3d49764c02d4d9437e59d35f8c49abb5bc94636c/src/compressed_tensors/transform/factory/base.py#L89-L91)

Reply:
Yes, we could handle this by calling `match_named_modules`
 separately for each target and compiling the list outside. I can implement that

Reply:
Okay, I've simplified the utility fn again, and I'm going to do the target matching logic outside, but I was considering making that into another utility function in `match.py`. i.e.

```python
def match_targets(
    name: str, module: torch.nn.Module, targets: Iterable[str]
) -> Generator[str]:
    """
    Yields the targets that match the given name and module.
    Outputs are ordered by type: exact name match, regex name match, class name match
    """
    targets = sorted(targets, key=lambda x: ("re:" in x, x))
    for target in targets:
        if _match_name(name, target):
            yield target

    for target in targets:
        if _match_class(module, target):
            yield target
```

Thoughts on that @kylesayrs?

Reply:
See https://github.com/neuralmagic/compressed-tensors/pull/406/commits/d22149f8585543fa139deb762beebf6601ffbcbf